components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java

```diff
@@ -92,13 +94,18 @@ private boolean isActionTypeEnabled(String actionTypePropertyName) {
     public Set<String> getExcludedHeadersInActionRequestForActionType(ActionType actionType) {
 
         Set<String> excludedHeaders = getExcludedHeadersInActionRequestForAllTypes();
-
+        List<String> excludedHeadersPropertyValue;
         switch (actionType) {
             case PRE_ISSUE_ACCESS_TOKEN:
-                List<String> excludedHeadersPropertyValue = getPropertyValues(
+                excludedHeadersPropertyValue = getPropertyValues(
                         ActionTypeConfig.PRE_ISSUE_ACCESS_TOKEN.getExcludedHeadersProperty());
                 excludedHeaders.addAll(excludedHeadersPropertyValue);
                 break;
+            case AUTHENTICATION:
+                excludedHeadersPropertyValue = getPropertyValues(
+                        ActionTypeConfig.AUTHENTICATION.getExcludedHeadersProperty());
+                excludedHeaders.addAll(excludedHeadersPropertyValue);
```

Comment:
Shall we move this out of the switch case

Reply:
Addressed with commit 2791587f98eb05e7560e1de7950fb41ca23d1218